File: Problems_and_Solutions/day-8/Java/Solution3.java
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class Solution3 {
    public static Node constructLL(int arr[]){
        Node head = new Node(arr[0]);
        Node temp = head;
        for(int i=1; i < arr.length; i++){
            Node newNode = new Node(arr[i]);
            temp.next = newNode;
            temp = newNode;
        }
        return head;
    }

    public static void printList(Node head){
        Node curr = head;
        while(curr != null){
            System.out.print(curr.data + " -> ");
            curr = curr.next;
        }
        System.out.println("null");
    }

    public static void main(String[] args){
        int[] arr = {1, 2, 3, 4, 5};
        Node head = constructLL(arr);

        System.out.println("Constructed Linked List:");
        printList(head);
    }
}

File: Problems_and_Solutions/day-8/cpp/solution 3.cpp
class Solution {
  public:
    Node* constructLL(vector<int>& arr) {
        // code here
        Node* head=NULL;
        Node* prev=NULL;
        Node* cur=NULL;
        for(int i:arr)
        {
            cur=new Node(i);
            if(prev!=NULL)
                prev->next=cur;
            if(head==NULL)
                head=cur;
            prev=cur;
        }
        return head;
    }
};
